Services/Mail/classes/Address/Type/class.ilMailRoleAddressType.php

```diff
@@ -179,18 +180,23 @@ public function resolve()
 	 * objects that are possible recipients for the role mailbox address.
 	 *
 	 * If Pear Mail is not installed, then the mailbox address
-	 * @param	string	IETF RFX 822 address list containing role mailboxes.
-	 * @return	int[] Array with role ids that were found
+	 * @param $a_address_list
+	 * @param null $dic
+	 * @param null $parser
+	 * @return int[] Array with role ids that were found
+	 * @internal param IETF $string RFX 822 address list containing role mailboxes.
 	 */
-	public static function searchRolesByMailboxAddressList($a_address_list)
+	public static function searchRolesByMailboxAddressList($a_address_list, ilMailRfc822AddressParserFactory $parser = null)
 	{
 		global $DIC;
 
-		$role_ids = array();
+		if ($parser === null) {
+			$parser = ilMailRfc822AddressParserFactory::getParser($a_address_list);
```

Comment:
One last thing. Maybe we should switch to an instance based approach of for this factory as well.

Reply:
Good point! I will open a seperated PR for this :+1: